src/content/docs/en/basics/astro-pages.mdx

```diff
@@ -96,10 +96,54 @@ Files with the `.html` file extension can be placed in the `src/pages/` director
 
 ## Custom 404 Error Page
 
-For a custom 404 error page, you can create a `404.astro` or `404.md` file in `/src/pages`.
+For a custom 404 error page, you can create a `404.astro` or `404.md` file in `src/pages`.
 
 This will build to a `404.html` page. Most [deploy services](/en/guides/deploy/) will find and use it.
 
+## Custom 500 Error Page
+
+For a custom 500 error page to show for pages in your `hybrid` or `server` project that are [rendered on demand](/en/guides/server-side-rendering/#enable-on-demand-server-rendering), create the file `src/pages/500.astro`. This custom page is not available for prerendered pages.
+
+If an error occurs rendering this page, your host's default 500 error page will be shown to your visitor.
+
+### Use the `error` prop
+
+<p><Since v="4.10.0" /></p>
+
+`src/pages/500.astro` is a special page that is automatically passed an `error` prop for any error thrown during rendering. This allows you to use the details of an error (e.g. from a page, from middleware, etc.) to display information to your visitor. 
+
+The error prop's data type can be anything, which may affect how you type or use the value in your code:
+
+```astro title="src/pages/500.astro"
+---
+interface Props {
+    error: unknown
+}
+
+const { error } = Astro.props
+---
+
+<div>{error instanceof Error ? error.message : 'Unknown error'}</div>
+```
+
+### Enable in development mode
+
+<p><Since v="4.10.0" /></p>
+
+You can use your custom 500 error page in development by setting the `ASTRO_CUSTOM_500` environment variable to `'true'`. This value can be included in a `.env` file:
+
+```ini title=".env"
+ASTRO_CUSTOM_500=true
+```
+
+It is also possible to use the CLI to set this environment variable:
+
+```sh
+ASTRO_CUSTOM_500=true astro dev
+```
+
+If an error occurs when rendering this custom page in development mode, Astro's default error overlay will be shown instead.
+
```

Comment:
``````suggestion
### Use the `error` prop

<p><Since v="4.11.0" /></p>

`src/pages/500.astro` is a special page that is automatically passed an `error` prop for any error thrown during rendering. This allows you to use the details of an error (e.g. from a page, from middleware, etc.) to display information to your visitor.

The error prop's data type can be anything, which may affect how you type or use the value in your code:

```astro title="src/pages/500.astro"
---
interface Props {
    error: unknown
}

const { error } = Astro.props
---

<div>{error instanceof Error ? error.message : 'Unknown error'}</div>
```

Be careful not to leak any sensitive information such as the error's stack.

``````

Reply:
> Be careful not to leak any sensitive information such as the error's stack.

Can we be more descriptive here? "Be careful not to..." but then no mention of how to avoid doing this. 😅 How would one do that? I'm envisioning something like...

> To avoid leaking sensitive information when displaying content from the `error` prop, ... 
and then fill in that blank. Whatever an appropriate action (or two) might be:

> ... consider evaluating the `error` prop first, and returning appropriate content based on the error thrown
 ???